cpp/daal/src/sycl/sum_reducer.cpp

```diff
@@ -207,12 +211,9 @@ SumReducer::Result SumReducer::sum(Layout vectorsLayout, const UniversalBuffer &
     const uint32_t maxWorkItemsPerGroup = 256;
     const uint32_t maxNumSubSlices      = 9;
 
-    Result result(context, nVectors, vectors.type(), status);
-    DAAL_CHECK_STATUS_RETURN_IF_FAIL(status, result);
-
     if (vectorsLayout == Layout::RowMajor)
     {
-        status |= sum_singlepass(context, kernelFactory, vectorsLayout, vectors, nVectors, vectorSize, maxWorkItemsPerGroup, result);
+        status |= sum_singlepass(context, kernelFactory, vectors, nVectors, vectorSize, result);
     }
     else
     {
```

Comment:
After this line there is a switch for defining numDivisionsByRow.
There is similar switch in LOM and i found out that these numbers showed best performance

    nRowsBlocks = 128;
    if (nVectors < 5000)
        nRowsBlocks = 1;
    else if (nVectors < 10000)
        nRowsBlocks = 8;
    else if (nVectors < 20000)
        nRowsBlocks = 16;
    else if (nVectors < 50000)
        nRowsBlocks = 32;
    else if (nVectors < 100000)
        nRowsBlocks = 64;

you can try it to see if it provides any benefit here

Reply:
Will add in a separate MR